File: src/Nethereum.Generators.JavaScript/codeGenNodeTest/StructSample/StructSampleService.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Numerics;
using Nethereum.Hex.HexTypes;
using Nethereum.ABI.FunctionEncoding.Attributes;
using Nethereum.Web3;
using Nethereum.RPC.Eth.DTOs;
using Nethereum.Contracts.CQS;
using Nethereum.Contracts.ContractHandlers;
using Nethereum.Contracts;
using System.Threading;
using Nethereum.Structs.StructSample.ContractDefinition;

namespace Nethereum.Structs.StructSample
{
    public partial class StructSampleService
    {
        public static Task<TransactionReceipt> DeployContractAndWaitForReceiptAsync(Nethereum.Web3.Web3 web3, StructSampleDeployment structSampleDeployment, CancellationTokenSource cancellationTokenSource = null)
        {
            return web3.Eth.GetContractDeploymentHandler<StructSampleDeployment>().SendRequestAndWaitForReceiptAsync(structSampleDeployment, cancellationTokenSource);
        }

        public static Task<string> DeployContractAsync(Nethereum.Web3.Web3 web3, StructSampleDeployment structSampleDeployment)
        {
            return web3.Eth.GetContractDeploymentHandler<StructSampleDeployment>().SendRequestAsync(structSampleDeployment);
        }

        public static async Task<StructSampleService> DeployContractAndGetServiceAsync(Nethereum.Web3.Web3 web3, StructSampleDeployment structSampleDeployment, CancellationTokenSource cancellationTokenSource = null)
        {
            var receipt = await DeployContractAndWaitForReceiptAsync(web3, structSampleDeployment, cancellationTokenSource);
            return new StructSampleService(web3, receipt.ContractAddress);
        }

        protected Nethereum.Web3.Web3 Web3{ get; }

        public ContractHandler ContractHandler { get; }

        public StructSampleService(Nethereum.Web3.Web3 web3, string contractAddress)
        {
            Web3 = web3;
            ContractHandler = web3.Eth.GetContractHandler(contractAddress);
        }

        public Task<GetTestOutputDTO> GetTestQueryAsync(GetTestFunction getTestFunction, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<GetTestFunction, GetTestOutputDTO>(getTestFunction, blockParameter);
        }

        public Task<GetTestOutputDTO> GetTestQueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<GetTestFunction, GetTestOutputDTO>(null, blockParameter);
        }

        public Task<string> SetStorageStructRequestAsync(SetStorageStructFunction setStorageStructFunction)
        {
             return ContractHandler.SendRequestAsync(setStorageStructFunction);
        }

        public Task<TransactionReceipt> SetStorageStructRequestAndWaitForReceiptAsync(SetStorageStructFunction setStorageStructFunction, CancellationTokenSource cancellationToken = null)
        {
             return ContractHandler.SendRequestAndWaitForReceiptAsync(setStorageStructFunction, cancellationToken);
        }

        public Task<string> SetStorageStructRequestAsync(TestStruct testStruct)
        {
            var setStorageStructFunction = new SetStorageStructFunction();
                setStorageStructFunction.TestStruct = testStruct;
            
             return ContractHandler.SendRequestAsync(setStorageStructFunction);
        }

        public Task<TransactionReceipt> SetStorageStructRequestAndWaitForReceiptAsync(TestStruct testStruct, CancellationTokenSource cancellationToken = null)
        {
            var setStorageStructFunction = new SetStorageStructFunction();
                setStorageStructFunction.TestStruct = testStruct;
            
             return ContractHandler.SendRequestAndWaitForReceiptAsync(setStorageStructFunction, cancellationToken);
        }

        public Task<string> TestRequestAsync(TestFunction testFunction)
        {
             return ContractHandler.SendRequestAsync(testFunction);
        }

        public Task<TransactionReceipt> TestRequestAndWaitForReceiptAsync(TestFunction testFunction, CancellationTokenSource cancellationToken = null)
        {
             return ContractHandler.SendRequestAndWaitForReceiptAsync(testFunction, cancellationToken);
        }

        public Task<string> TestRequestAsync(TestStruct testScrut)
        {
            var testFunction = new TestFunction();
                testFunction.TestScrut = testScrut;
            
             return ContractHandler.SendRequestAsync(testFunction);
        }

        public Task<TransactionReceipt> TestRequestAndWaitForReceiptAsync(TestStruct testScrut, CancellationTokenSource cancellationToken = null)
        {
            var testFunction = new TestFunction();
                testFunction.TestScrut = testScrut;
            
             return ContractHandler.SendRequestAndWaitForReceiptAsync(testFunction, cancellationToken);
        }

        public Task<TestArrayOutputDTO> TestArrayQueryAsync(TestArrayFunction testArrayFunction, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<TestArrayFunction, TestArrayOutputDTO>(testArrayFunction, blockParameter);
        }

        public Task<TestArrayOutputDTO> TestArrayQueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<TestArrayFunction, TestArrayOutputDTO>(null, blockParameter);
        }

        public Task<BigInteger> Id1QueryAsync(Id1Function id1Function, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id1Function, BigInteger>(id1Function, blockParameter);
        }

        
        public Task<BigInteger> Id1QueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id1Function, BigInteger>(null, blockParameter);
        }

        public Task<BigInteger> Id2QueryAsync(Id2Function id2Function, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id2Function, BigInteger>(id2Function, blockParameter);
        }

        
        public Task<BigInteger> Id2QueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id2Function, BigInteger>(null, blockParameter);
        }

        public Task<BigInteger> Id3QueryAsync(Id3Function id3Function, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id3Function, BigInteger>(id3Function, blockParameter);
        }

        
        public Task<BigInteger> Id3QueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id3Function, BigInteger>(null, blockParameter);
        }

        public Task<string> Id4QueryAsync(Id4Function id4Function, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id4Function, string>(id4Function, blockParameter);
        }

        
        public Task<string> Id4QueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryAsync<Id4Function, string>(null, blockParameter);
        }

        public Task<TestStructStorageOutputDTO> TestStructStorageQueryAsync(TestStructStorageFunction testStructStorageFunction, BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<TestStructStorageFunction, TestStructStorageOutputDTO>(testStructStorageFunction, blockParameter);
        }

        public Task<TestStructStorageOutputDTO> TestStructStorageQueryAsync(BlockParameter blockParameter = null)
        {
            return ContractHandler.QueryDeserializingToObjectAsync<TestStructStorageFunction, TestStructStorageOutputDTO>(null, blockParameter);
        }
    }
}

File: src/Nethereum.Generators.JavaScript/codeGenNodeTest/StructSample/StructSampleService.fs
namespace Nethereum.Structs.StructSample

open System
open System.Threading.Tasks
open System.Collections.Generic
open System.Numerics
open Nethereum.Hex.HexTypes
open Nethereum.ABI.FunctionEncoding.Attributes
open Nethereum.Web3
open Nethereum.RPC.Eth.DTOs
open Nethereum.Contracts.CQS
open Nethereum.Contracts.ContractHandlers
open Nethereum.Contracts
open System.Threading
open Nethereum.Structs.StructSample.ContractDefinition


    type StructSampleService (web3: Web3, contractAddress: string) =
    
        member val Web3 = web3 with get
        member val ContractHandler = web3.Eth.GetContractHandler(contractAddress) with get
    
        static member DeployContractAndWaitForReceiptAsync(web3: Web3, structSampleDeployment: StructSampleDeployment, ?cancellationTokenSource : CancellationTokenSource): Task<TransactionReceipt> = 
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            web3.Eth.GetContractDeploymentHandler<StructSampleDeployment>().SendRequestAndWaitForReceiptAsync(structSampleDeployment, cancellationTokenSourceVal)
        
        static member DeployContractAsync(web3: Web3, structSampleDeployment: StructSampleDeployment): Task<string> =
            web3.Eth.GetContractDeploymentHandler<StructSampleDeployment>().SendRequestAsync(structSampleDeployment)
        
        static member DeployContractAndGetServiceAsync(web3: Web3, structSampleDeployment: StructSampleDeployment, ?cancellationTokenSource : CancellationTokenSource) = async {
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            let! receipt = StructSampleService.DeployContractAndWaitForReceiptAsync(web3, structSampleDeployment, cancellationTokenSourceVal) |> Async.AwaitTask
            return new StructSampleService(web3, receipt.ContractAddress);
            }
    
        member this.OwnersQueryAsync(ownersFunction: OwnersFunction, ?blockParameter: BlockParameter): Task<string> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<OwnersFunction, string>(ownersFunction, blockParameterVal)
            
        member this.SubmitTransactionRequestAsync(submitTransactionFunction: SubmitTransactionFunction): Task<string> =
            this.ContractHandler.SendRequestAsync(submitTransactionFunction);
        
        member this.SubmitTransactionRequestAndWaitForReceiptAsync(submitTransactionFunction: SubmitTransactionFunction, ?cancellationTokenSource : CancellationTokenSource): Task<TransactionReceipt> =
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            this.ContractHandler.SendRequestAndWaitForReceiptAsync(submitTransactionFunction, cancellationTokenSourceVal);
        
        member this.RevokeConfirmationRequestAsync(revokeConfirmationFunction: RevokeConfirmationFunction): Task<string> =
            this.ContractHandler.SendRequestAsync(revokeConfirmationFunction);
        
        member this.RevokeConfirmationRequestAndWaitForReceiptAsync(revokeConfirmationFunction: RevokeConfirmationFunction, ?cancellationTokenSource : CancellationTokenSource): Task<TransactionReceipt> =
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            this.ContractHandler.SendRequestAndWaitForReceiptAsync(revokeConfirmationFunction, cancellationTokenSourceVal);
        
        member this.IsOwnerQueryAsync(isOwnerFunction: IsOwnerFunction, ?blockParameter: BlockParameter): Task<bool> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<IsOwnerFunction, bool>(isOwnerFunction, blockParameterVal)
            
        member this.ConfirmationsQueryAsync(confirmationsFunction: ConfirmationsFunction, ?blockParameter: BlockParameter): Task<bool> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<ConfirmationsFunction, bool>(confirmationsFunction, blockParameterVal)
            
        member this.GetTransactionCountQueryAsync(getTransactionCountFunction: GetTransactionCountFunction, ?blockParameter: BlockParameter): Task<BigInteger> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<GetTransactionCountFunction, BigInteger>(getTransactionCountFunction, blockParameterVal)
            
        member this.IsConfirmedQueryAsync(isConfirmedFunction: IsConfirmedFunction, ?blockParameter: BlockParameter): Task<bool> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<IsConfirmedFunction, bool>(isConfirmedFunction, blockParameterVal)
            
        member this.GetConfirmationCountQueryAsync(getConfirmationCountFunction: GetConfirmationCountFunction, ?blockParameter: BlockParameter): Task<BigInteger> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<GetConfirmationCountFunction, BigInteger>(getConfirmationCountFunction, blockParameterVal)
            
        member this.TransactionsQueryAsync(transactionsFunction: TransactionsFunction, ?blockParameter: BlockParameter): Task<TransactionsOutputDTO> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryDeserializingToObjectAsync<TransactionsFunction, TransactionsOutputDTO>(transactionsFunction, blockParameterVal)
            
        member this.GetOwnersQueryAsync(getOwnersFunction: GetOwnersFunction, ?blockParameter: BlockParameter): Task<List<string>> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<GetOwnersFunction, List<string>>(getOwnersFunction, blockParameterVal)
            
        member this.GetTransactionIdsQueryAsync(getTransactionIdsFunction: GetTransactionIdsFunction, ?blockParameter: BlockParameter): Task<List<BigInteger>> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<GetTransactionIdsFunction, List<BigInteger>>(getTransactionIdsFunction, blockParameterVal)
            
        member this.GetConfirmationsQueryAsync(getConfirmationsFunction: GetConfirmationsFunction, ?blockParameter: BlockParameter): Task<List<string>> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<GetConfirmationsFunction, List<string>>(getConfirmationsFunction, blockParameterVal)
            
        member this.TransactionCountQueryAsync(transactionCountFunction: TransactionCountFunction, ?blockParameter: BlockParameter): Task<BigInteger> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<TransactionCountFunction, BigInteger>(transactionCountFunction, blockParameterVal)
            
        member this.ConfirmTransactionRequestAsync(confirmTransactionFunction: ConfirmTransactionFunction): Task<string> =
            this.ContractHandler.SendRequestAsync(confirmTransactionFunction);
        
        member this.ConfirmTransactionRequestAndWaitForReceiptAsync(confirmTransactionFunction: ConfirmTransactionFunction, ?cancellationTokenSource : CancellationTokenSource): Task<TransactionReceipt> =
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            this.ContractHandler.SendRequestAndWaitForReceiptAsync(confirmTransactionFunction, cancellationTokenSourceVal);
        
        member this.MAX_OWNER_COUNTQueryAsync(mAX_OWNER_COUNTFunction: MAX_OWNER_COUNTFunction, ?blockParameter: BlockParameter): Task<BigInteger> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<MAX_OWNER_COUNTFunction, BigInteger>(mAX_OWNER_COUNTFunction, blockParameterVal)
            
        member this.RequiredQueryAsync(requiredFunction: RequiredFunction, ?blockParameter: BlockParameter): Task<BigInteger> =
            let blockParameterVal = defaultArg blockParameter null
            this.ContractHandler.QueryAsync<RequiredFunction, BigInteger>(requiredFunction, blockParameterVal)
            
        member this.ExecuteTransactionRequestAsync(executeTransactionFunction: ExecuteTransactionFunction): Task<string> =
            this.ContractHandler.SendRequestAsync(executeTransactionFunction);
        
        member this.ExecuteTransactionRequestAndWaitForReceiptAsync(executeTransactionFunction: ExecuteTransactionFunction, ?cancellationTokenSource : CancellationTokenSource): Task<TransactionReceipt> =
            let cancellationTokenSourceVal = defaultArg cancellationTokenSource null
            this.ContractHandler.SendRequestAndWaitForReceiptAsync(executeTransactionFunction, cancellationTokenSourceVal);
        
    

